src/main/java/harmonised/pmmo/core/Core.java

```diff
@@ -454,26 +456,29 @@ public void getSalvage(ServerPlayer player) {
 		entry:
 		for (Map.Entry<ResourceLocation, SalvageData> result : loader.ITEM_LOADER.getData(RegistryUtil.getId(salvageItem)).salvage().entrySet()) {
 			//First look for any skills that do not meet the req and continue to the next output 
-			//item if the req is not met. 
+			//item if the req is not met.
 			for (Map.Entry<String, Integer> skill : result.getValue().levelReq().entrySet()) {
-				if (skill.getValue() > Core.get(LogicalSide.SERVER).getData().getLevelFromXP(playerXp.getOrDefault(skill.getKey(), 0l))) continue entry;
+				if (skill.getValue() > Core.get(LogicalSide.SERVER).getData().getLevelFromXP(playerXp.getOrDefault(skill.getKey(), 0L))) continue entry;
 			}
 			//ensures that only salvage where the reqs have been met AND the item has entries result in item consumption
 			validAttempt = true;
 			//get the base calculation values including the bonuses from skills
-			double base = result.getValue().baseChance();
-			double max = result.getValue().maxChance();
+			SalvageEvent salvageEvent = new SalvageEvent(player, salvageItem, result);
+			double base = salvageEvent.getSalvage().getValue().baseChance();
```

Comment:
if we go the route of making the event use builders, we may want to update the `getSalvage()` call to call `Builder#build` internally.  just a note.